### ISSUE TITLE
EASY: clean up meta example filenames

### ISSUE THREAD

**Issue by karlnapf** [shogun-toolbox/shogun#4050]
folders:
`gaussian_processes` -> `gaussian_process`
`multiclass_classifier` -> `multiclass`
`binary_classifier` -> `binary`

no folder names inside the program listing names, e.g.
`gaussian_process_classifier` -> `classifier`
`multiclass_linearmachine` -> `linear`
`sparse_gaussian_process_regression` -> `sparse_regression`
`multiclass_logisticregression` -> `logistic_regression`

no abbreviations
`svm` -> `support_vector_machine`
`knn` -> `k_nearest_neighbors`

etc.

Make sure to change the cookbook filenames in `doc/cookbook` as well (try your changes with `make cookbook`)

Good way to learn about the examples in Shogun while cleaning up a bit

**Comment by dgkim5360** [comment]
Hello, newbie here. May I work on this one? I managed to compile and install it on my Ubuntu machine, and it looks appropriate for me to do the first steps.

Thanks for the beginner-friendly issues :+1:

**Comment by karlnapf** [comment]
sure go for it :)

**Comment by dgkim5360** [comment]
@karlnapf while working on the cookbook, I found some minor typos, e.g.,
in `/doc/cookbook/source/examples/multiclass/k_nearest_neighbours.rst`

```diff
diff --git a/doc/cookbook/source/examples/multiclass/k_nearest_neighbours.rst b/doc/cookbook/source/examples/multiclass/k_nearest_neighbours.rst
index 58d6c6f..b0888ce 100644
--- a/doc/cookbook/source/examples/multiclass/k_nearest_neighbours.rst
+++ b/doc/cookbook/source/examples/multiclass/k_nearest_neighbours.rst
@@ -8,7 +8,7 @@ For :math:`k=1`, the label for a test point :math:`x^*` is predicted to be the s
 
 .. math::
 
-   k=\argmin_j d(x^*, x_j).  
+   k=\arg\min_j d(x^*, x_j).  
    
 See Chapter 14 in :cite:`barber2012bayesian` for a detailed introduction.
```
Should I include such corrections, or just leave them untouched and commit for the original task?

**Comment by karlnapf** [comment]
Argmin should be defined. Check the cookbook readme it tell you where to put custom latex

**Comment by karlnapf** [comment]
If you find typos, make a separate pr

**Comment by shivampkumar** [comment]
Just starting out with Shogun. I will go with this one first...Is that cool?

**Comment by karlnapf** [comment]
Closed via #4055
